From d179c4684e2aa04041110abe2c8c84cd6a148e98 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Fri, 26 Jul 2019 17:27:56 +0530
Subject: [PATCH 113/138] PCI: quirks : Apply ACS quirk for all devices

commit 834ff2008af97f493b5e066b993e628d58e47ff0 from
git@git.assembla.com:cavium/WindRiver.linux.git

Due to the current quirk ACS capability is set to only for
the root devices.
Multifunction devices get the same iommu group number if ACS
capability is not set. Hence, changing quirk to set ACS capability
for all devices.

Change-Id: Idff6c92229e5e568db5bc19efce68f983d3d59dc
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/13115
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/13138
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/pci/quirks.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 70366add1794..fd2e93b2619f 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -4105,9 +4105,7 @@ static bool pci_quirk_cavium_acs_match(struct pci_dev *dev)
 	 * bits of device ID are used to indicate which subdevice is used
 	 * within the SoC.
 	 */
-	return (pci_is_pcie(dev) &&
-		(pci_pcie_type(dev) == PCI_EXP_TYPE_ROOT_PORT) &&
-		((dev->device & 0xf800) == 0xa000));
+	return (pci_is_pcie(dev) && ((dev->device & 0xf800) == 0xa000));
 }
 
 static int pci_quirk_cavium_acs(struct pci_dev *dev, u16 acs_flags)
-- 
2.17.1

